test/functional-tests/include/TmpFile.hpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include "Exception.hpp"
+
+#include <exception>
+#include <fstream>
+#include <string>
+#include <array>
+
+#include <cerrno>
+#include <cstring>
+
+class TmpFile {
+public:
+    TmpFile(std::string content) : mPath(mktmp())
+    {
+        std::ofstream file(mPath);
+        file.exceptions(std::ofstream::failbit | std::ofstream::badbit);
+        std::ofstream(mPath) << content;
+    }
+    TmpFile(TmpFile &&right) :
+        mPath(std::move(right.mPath))
+    {
+        right.mPath="";
+    }
+    ~TmpFile()
+    {
+        if (not mPath.empty()) {
+            std::remove(mPath.c_str());
+        }
+    }
+    
+    const std::string &getPath() { return mPath; }
+private:
+    std::string mktmp() {
+        std::array<char, L_tmpnam> buffer;
+        char * res = std::tmpnam(buffer.data());
+        if (res == nullptr) {
+            throw Exception("Could not create tmp file: " + strerror());
```

Comment:
I wanted to differentiate between Errors from the wrapper and errors from the underline library (as it is not supposed to throw).